### PR TITLE
Cleanup align handling

### PR DESCRIPTION
**Pull request by ffreyer**
# Description

Mostly pulled out of #2944.

We have a lot of places that repeat the symbol -> number conversion (and many that don't accept numbers directly) which I think would be good to handle centrally and equally. This pr introduces `halign2num` and `valign2num` for that and simplifies the affected code to function calls.

<details><summary> Discarded TODO</summary>

Should justification inherit numeric haligns? I.e. something like `halign = 0.2`?
https://github.com/MakieOrg/Makie.jl/blob/3d162f58d76f5338980dfa570f34da505d14379d/src/layouting/layouting.jl#L172-L181

Given that PolarAxis' angle -> align conversion has spread to barplot now, should we add an official conversion for it? Or is that too much of a niche case?
https://github.com/MakieOrg/Makie.jl/blob/3d162f58d76f5338980dfa570f34da505d14379d/src/basic_recipes/barplot.jl#L175-L177
</details>

## Type of change

- internal improvement

## Checklist

- [ ] Added an entry in NEWS.md (for new features and breaking changes)
- [ ] Added or changed relevant sections in the documentation
- [x] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

**Comment by ffreyer** [comment]
> Should justification inherit numeric haligns? I.e. something like `halign = 0.2`?
> 
> https://github.com/MakieOrg/Makie.jl/blob/3d162f58d76f5338980dfa570f34da505d14379d/src/layouting/layouting.jl#L172-L181

Old (master) vs new (allowing floats in halign to pass) justification

Screenshot from 2023-08-29 18-58-04

**Comment by jkrumbiegel** [comment]
> Should justification inherit numeric haligns?

I don't think so, halfway justification looks weird to me :)

**Comment by ffreyer** [comment]
Ok I think Adding `text(..., align = some_angle)` is a bit weird as well, and it requires more than just cleanup. So let's drop that and justification changes from this.

If tests pass we can merge this.

**Comment by jkrumbiegel** [comment]
Does baseline valign still work?

File: src/layouting/layouting.jl [hunk 1]
```diff
@@ -233,17 +217,7 @@ function glyph_collection(
...
-        elseif valign === :bottom
-            0f0
-        elseif valign === :center
-            0.5f0
-        else
-            error("Invalid valign $valign. Valid values are <:Number, :bottom, :baseline, :top, and :center.")
-        end
+        va = valign2num(valign, "Invalid valign $valign. Valid values are <:Number, :bottom, :baseline, :top, and :center.")
         ys .- first_line_ascender .+ (1 - va) .* overall_height
     end
```
[the remark this answers is not in the document]
Comment (jkrumbiegel) [reply src/layouting/layouting.jl:222]: ah right it handles that before it goes into `valign2num`, all right

**Review by jkrumbiegel — APPROVED** [review]

**Pull request merged**